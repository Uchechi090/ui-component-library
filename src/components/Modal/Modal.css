.modal-wrapper {
  position: fixed;
  left: 0;
  top: 0;
  height: 100%;
  width: 100%;
  z-index: 1000;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.overlay {
  position: absolute;
  top: 0;
  left: 0;
  bottom: 0;
  right: 0;
  width: 100%;
  height: 100%;
  background-color: rgb(107 114 128);
  opacity: 0.75;
}

.modal {
  border-radius: 8px;
  opacity: 1;
  z-index: 1001;
  padding: 24px 20px;
  background-color: #fff;
  box-shadow: 0 1px 3px 0 rgb(0 0 0 / 0.1), 0 1px 2px -1px rgb(0 0 0 / 0.1);
  width: 450px;
  min-width: 350px;
  max-width: 1100px;
  position: relative;
  bottom: 190px;
  overflow-y: auto;
  overflow-x: hidden;
  box-sizing: border-box;
}

.modal-heading {
  display: flex;
  margin-bottom: 28px;
}

h3 {
  color: #C53678;
  font-size: 24px;
  line-height: 16px;
  font-weight: 500;
  margin: 0;
}

.close {
  position: absolute;
  right: 11px;
  top: 11px;
  z-index: 1004;
  cursor: pointer;
}

.modal-footer {
  border-width: 0px;
  border-top-width: 1px;
  border-color: rgb(229 231 235);
  display: flex;
  justify-content: flex-end;
  padding-top: 15px;
}
