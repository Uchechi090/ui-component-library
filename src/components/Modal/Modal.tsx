import React from "react";
import { GrFormClose } from "react-icons/gr";

import "./Modal.css";

interface ModalProps {
  isOpen: boolean;
  onClose: React.Dispatch<React.SetStateAction<boolean>>;
}

const Modal: React.FC<ModalProps> = ({ isOpen, onClose }) => {
  return (
    <div data-testid="modal-wrapper" className="modal-wrapper">
      <div data-testid="overlay" className="overlay" onClick={() => onClose(false)} />
      <div className="modal">
        <div className="modal-heading">
          <h3>Custom Modal</h3>
          <div
            className="close"
            data-testid="close-button"
            onClick={() => onClose(false)}
          >
            <GrFormClose />
          </div>
        </div>
        <div className="modal-content">
          <p>
            Lorem ipsum dolor sit amet consectetur adipisicing elit. Distinctio
            a at eaque atque impedit, voluptates doloribus deleniti consequatur
            modi nemo cupiditate eligendi ipsam voluptatem sint maxime explicabo
            alias. Eligendi, pariatur.
          </p>
          {isOpen && <p>Modal is open</p>}
        </div>
        <div className="modal-footer"></div>
      </div>
    </div>
  );
};

export default Modal;
