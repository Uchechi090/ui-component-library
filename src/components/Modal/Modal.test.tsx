import React from "react";
import { render, fireEvent } from "@testing-library/react";

import Modal from "./Modal";

describe("<Modal />", () => {
  const mockCloseValue = jest.fn();
  let isOpen = true;

  it("should show props", () => {
    const { getByTestId, asFragment, rerender } = render(
      <Modal isOpen={isOpen} onClose={mockCloseValue} />
    );

    expect(asFragment()).toMatchSnapshot();
    expect(getByTestId("modal-wrapper")).toBeInTheDocument();

    fireEvent.click(getByTestId("close-button"));
    expect(mockCloseValue).toBeCalled();

    rerender(<Modal isOpen={isOpen} onClose={mockCloseValue} />);

    fireEvent.click(getByTestId("overlay"));
    expect(mockCloseValue).toBeCalled();
  });
});
