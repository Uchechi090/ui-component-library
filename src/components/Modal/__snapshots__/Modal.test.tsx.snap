// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> should show props 1`] = `
<DocumentFragment>
  <div
    class="modal-wrapper"
    data-testid="modal-wrapper"
  >
    <div
      class="overlay"
      data-testid="overlay"
    />
    <div
      class="modal"
    >
      <div
        class="modal-heading"
      >
        <h3>
          Custom Modal
        </h3>
        <div
          class="close"
          data-testid="close-button"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M7,7 L17,17 M7,17 L17,7"
              fill="none"
              stroke="#000"
              stroke-width="2"
            />
          </svg>
        </div>
      </div>
      <div
        class="modal-content"
      >
        <p>
          Lorem ipsum dolor sit amet consectetur adipisicing elit. Distinctio a at eaque atque impedit, voluptates doloribus deleniti consequatur modi nemo cupiditate eligendi ipsam voluptatem sint maxime explicabo alias. Eligendi, pariatur.
        </p>
        <p>
          Modal is open
        </p>
      </div>
      <div
        class="modal-footer"
      />
    </div>
  </div>
</DocumentFragment>
`;
